primitives/core/src/crypto.rs

```diff
@@ -281,7 +281,10 @@ pub trait Ss58Codec: Sized + AsMut<[u8]> + AsRef<[u8]> + Default {
 			// Invalid length.
 			return Err(PublicError::BadLength);
 		}
-		let ver = d[0].try_into().map_err(|_: ()| PublicError::UnknownVersion)?;
+		let ver = d[0]
+			.try_into()
+			.or_else(|_| d[0].to_string().as_str().try_into())
```

Comment:
I think the more appropriate solution would be to fix the `TryFrom<u8>` for `Ss58AddressFormat`.

If a number is unknown, we interpret it as custom number.

Reply:
I meet this problem in another chain again. I think this may be changed to fix this part:
```rust
impl TryFrom<u8> for Ss58AddressFormat {
			type Error = ();

			fn try_from(x: u8) -> Result<Ss58AddressFormat, ()> {
				match x {
					$($number => Ok(Ss58AddressFormat::$identifier)),*,
					_ => Err(()),
				}
			}
		}

		impl<'a> TryFrom<&'a str> for Ss58AddressFormat {
			type Error = ();

			fn try_from(x: &'a str) -> Result<Ss58AddressFormat, ()> {
				match x {
					$($name => Ok(Ss58AddressFormat::$identifier)),*,
					a => a.parse::<u8>().map(Ss58AddressFormat::Custom).map_err(|_| ()),
				}
			}
		}
```
may be we should add custem match for this macro like:
```
impl TryFrom<u8> for Ss58AddressFormat {
			type Error = ();

			fn try_from(x: u8) -> Result<Ss58AddressFormat, ()> {
				match x {
					$($number => Ok(Ss58AddressFormat::$identifier)),*,
					_ => {
						let custom = match *DEFAULT_VERSION.lock() {
							Ss58AddressFormat::Custom(n) => n,
							_ => return Err(())
						};
						if custom != x {
							return Err(())
						}
						Ss58AddressFormat::Custom(x)
					},
				}
			}
		}
```

But I'm sure, if someone set `Ss58AddressFormat::Custom(n)` which n within in existed Ss58AddressFormat list, would cause error...
e.g. some one set 45 as Custom version, but now 45 have not in existed Ss58AddressFormat. it works fun. However someday, an another chain create a pr to set 45 as their chain ss58AddressFormat address, for example add
```rust
Chain45 => (45, "reserved45", "Reserved for future use (45).")
```
than if the old chain use same code, do `try_from` would generate :
```rust
Ss58AddressFormat::Chain45
```
not
```rust
Ss58AddressFormat::Custom(45)
```

Reply:
In another word, this change would meet the code like:
```rust
// if set 42 as default version rather than `Ss58AddressFormat::SubstrateAccount`
set_default_ss58_version(Ss58AddressFormat::Custom(42))
```
than if do a valid address like:
```rust
// this addr is 42 version
let addr = "KWAfgC2aRG5UVD6CpbPQXCx4YZZUhvWqqAJE6qcYc9Rtr6g5C";
let (pub, ver) = Public::from_ss58check_with_version(&addr)
assert(ver, Ss58AddressFormat::SubstrateAccount); // pass
assert(ver, Ss58AddressFormat::Custom(42)); // not pass
```
this is just an exmpale, I mean, no one would use 42 as custom, but some one may use another reversed version(for exmpale 45), than one day, a team or chain developer would occupy 45 as their version, than the old one may cause problem.